constructor/conda_interface.py

```diff
@@ -52,26 +53,36 @@
 
     from conda.exports import cache_fn_url as _cache_fn_url
 
-    def write_repodata(cache_dir, url):
+    def write_repodata(cache_dir, url, used_packages):
         if CONDA_MAJOR_MINOR >= (4, 5):
             from conda.core.subdir_data import fetch_repodata_remote_request
             raw_repodata_str = fetch_repodata_remote_request(url, None, None)
             repodata_filename = _cache_fn_url(url)
-            with open(join(cache_dir, repodata_filename), 'w') as fh:
-                fh.write(raw_repodata_str)
         elif CONDA_MAJOR_MINOR >= (4, 4):
             from conda.core.repodata import fetch_repodata_remote_request
             raw_repodata_str = fetch_repodata_remote_request(url, None, None)
             repodata_filename = _cache_fn_url(url)
-            with open(join(cache_dir, repodata_filename), 'w') as fh:
-                fh.write(raw_repodata_str)
         elif CONDA_MAJOR_MINOR >= (4, 3):
             from conda.core.repodata import fetch_repodata_remote_request
             repodata_obj = fetch_repodata_remote_request(None, url, None, None)
             raw_repodata_str = json.dumps(repodata_obj)
             repodata_filename = _cache_fn_url(url)
-            with open(join(cache_dir, repodata_filename), 'w') as fh:
-                fh.write(raw_repodata_str)
         else:
             raise NotImplementedError("unsupported version of conda: %s" % CONDA_INTERFACE_VERSION)
+        full_repodata = json.loads(raw_repodata_str)
+        used_repodata = {k: full_repodata[k] for k in set(full_repodata.keys()) - set(('packages',
+                                                                                       'packages.conda',
+                                                                                       'removed',))}
+        used_repodata['packages'] = {}
+        used_repodata['packages.conda'] = {}
+        used_repodata['removed'] = []
+        # arbitrary old, expired date, so that conda will want to immediately update it
+        # when not being run in offline mode
+        used_repodata['_mod'] = "Mon, 07 Jan 2019 15:22:15 GMT"
+        for package in used_packages:
+            for key in ('packages', 'packages.conda'):
+                if package in full_repodata[key]:
```

Comment:
It looks like this will break if any channel is used that doesn't have the `packages.conda` key which happens with the r channel, etc.

Reply:
ah, good catch.  That should be a .get().